src/connection.ts

```diff
@@ -0,0 +1,66 @@
+import { Connection }  from '@libp2p/interface-connection'
+import {ConnectionStat}from '@libp2p/interface-connection'
+import { Stream, Direction } from '@libp2p/interface-connection'
+import { PeerId }  from '@libp2p/interface-peer-id';
+import { AbortOptions }from '@libp2p/interfaces';
+import { logger }      from '@libp2p/logger'
+import { Multiaddr }   from '@multiformats/multiaddr';
+import {v4 as genUuid} from 'uuid';
+
+const log = logger('libp2p:webrtc:connection')
+
+type ConnectionInit = {
+    id: string
+    localPeer: PeerId
+    localAddr?: Multiaddr
+    remotePeer: PeerId
+    remoteAddr: Multiaddr
+    direction: Direction
+    tags?: string[]
+    stat: ConnectionStat
+    pc:  RTCPeerConnection
+    credential_string: string
+}
+
+export class WebRTCConnection implements Connection {
+    id: string;
```

Comment:
Consider adding access modifiers here.

Reply:
What is the proper convention for data members that are satisfying some interface?

No modifier was present in the code I stole this from. Perhaps I should specify `public` ?

Reply:
I don't think there is a convention for separating out data members, though I've seen the sorted by access modification type in some repos.

You can be explicit about `public` or leave it off, it's stylistic.  If they're not meant to be used outside of this class, consider making all `private` until they need to be `public`.

Reply:
I think they can't be private. And given that, sorting by access modifier accomplishes what I was after.

````
src/connection.ts:25:14 - error TS2420: Class 'WebRTCConnection' incorrectly implements interface 'Connection'.
  Property 'id' is private in type 'WebRTCConnection' but not in type 'Connection'.

25 export class WebRTCConnection implements Connection {
````